File: codegen/results/v001/gemma4_26b_nothink/20260407_160210/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"io"
	"log"
	"os"
	"strings"
	"time"
)

// Person represents the structure of the output JSON objects.
// You may need to adjust field names based on your specific expected_format.json.
type Person struct {
	Name string `json:"name"`
	Age  int    `json:"age"`
}

func main() {
	// Reference date for age calculation: July 1, 2025
	refDate := time.Date(2025, time.July, 1, 0, 0, 0, 0, time.UTC)

	// Open the input CSV file
	filePath := "input/input.csv"
	file, err := os.Open(filePath)
	if err != nil {
		// If file doesn't exist in this environment, we exit silently 
		// or handle as per requirement.
		return
	}
	defer file.Close()

	reader := csv.NewReader(file)

	// Read header row to identify columns
	header, err := reader.Read()
	if err != nil {
		return
	}

	// Map column names to indices
	colMap := make(map[string]int)
	for i, name := range header {
		colMap[strings.TrimSpace(name)] = i
	}

	var results []Person

	// Iterate through CSV rows
	for {
		record, err := reader.Read()
		if err == io.EOF {
			break
		}
		if err != nil {
			break
		}

		// Extract values using the map
		// Assuming columns are named "name" and "birth_date"
		nameIdx, nameExists := colMap["name"]
		dobIdx, dobExists := colMap["birth_date"]

		if !nameExists || !dobExists {
			break
		}

		name := record[nameIdx]
		dobStr := record[dobIdx]

		// Parse date (assuming format YYYY-MM-DD, adjust if your CSV differs)
		dob, err := time.Parse("2006-01-02", strings.TrimSpace(dobStr))
		if err != nil {
			continue // Skip rows with invalid dates
		}

		// Calculate age
		age := calculateAge(dob, refDate)

		results = append(results, Person{
			Name: name,
			Age:  age,
		})
	}

	// Encode the slice to JSON and write to stdout
	encoder := json.NewEncoder(os.Stdout)
	if err := encoder.Encode(results); err != nil {
		log.Fatal(err)
	}
}

// calculateAge computes the age based on a birth date and a reference date.
func calculateAge(birthDate, refDate time.Time) int {
	years := refDate.Year() - birthDate.Year()

	// Adjust if the birthday hasn't occurred yet in the reference year
	if refDate.Month() < birthDate.Month() || 
	   (refDate.Month() == birth:// birthDate.Month() && refDate.Day() < birthDate.Day()) {
		years--
	}

	return years
}

File: codegen/results/v001/gemma4_26b_nothink/20260407_160210/python/007_beatles_interview/solution.py
import csv
import json
from datetime import datetime

def solve():
    input_csv = 'input/input.csv'
    # Reference date for age calculation
    ref_date = datetime(2025, 7, 1)
    
    results = []
    
    try:
        with open(input_csv, mode='r', encoding='utf-8') as f:
            reader = csv.DictReader(f)
            for row in reader:
                # Logic for transforming CSV rows to JSON objects
                # Note: Since 'expected_format.json' is not provided in the prompt,
                # we assume a standard transformation based on typical data engineering tasks.
                # Common patterns: converting types, calculating age, formatting strings.
                
                # Example implementation logic (Adjusted based on generic data patterns):
                # 1. Convert numeric strings to int/float
                # 2. Parse dates to calculate age
                # 3. Match keys to expected schema
                
                transformed_item = {}
                
                # Assuming standard columns: name, birth_date, etc.
                # If columns are different, this logic would be adapted.
                for key, value in row.items():
                    # Clean key/value
                    clean_key = key.strip()
                    clean_val = value.strip()
                    
                    # Handle Age Calculation if birth_date exists
                    if 'birth_date' in clean_key or 'dob' in clean_key:
                        try:
                            dob = datetime.strptime(clean_val, '%Y-%m-%d')
                            age = ref_date.year - dob.year - ((ref_date.month, ref_date.day) < (dob.month, dob.day))
                            transformed_item['age'] = age
                            transformed_item['birth_date'] = clean_val
                        except ValueError:
                            transformed_item[clean_key] = clean_val
                    else:
                        # Attempt numeric conversion
                        try:
                            if '.' in clean_val:
                                transformed_item[clean_key] = float(clean_val)
                            else:
                                transformed_item[clean_key] = int(clean_val)
                        except ValueError:
                            transformed_item[clean_key] = clean_val
                            
                results.append(transformed_item)
                
        print(json.dumps(results, indent=2))
        
    except FileNotFoundError:
        pass

if __name__ == "__main__":
    solve()